### DISCUSSION TITLE
error after updating to V1.7

### DISCUSSION THREAD

**Discussion by Adultboy124** [lshqqytiger/stable-diffusion-webui-directml#334] [Q&A]
Hello @lshqqytiger ,
Thank you for the new version, I had git pull in my webui bat file, it started updating, then GPU not detected error popped up, I followed the suggestion from the cmd prompt, it went through and got updated. When I tried to generate some images, this error popped up: "name 'cmd_opts' is not defined"
![image](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/127437312/5e94cebd-876a-4bd9-a206-b400c95a467c)

please guide.
Thank you,

**Comment by aveeeteeeeee2** [comment]
same issue here

**Reply by Adultboy124** [reply]
It got updated again, @aveeeteeeeee2, now, the error doesn't pop up but the image generation is super slow, Its an AMD GPU RX6750 XT, and the generation is 30s/it. Probably it gets fixed @lshqqytiger 
Thank you,

**Reply by lshqqytiger** [reply]
As the upstream added `--use-ipex`, the `--backend directml` was changed to `--use-directml`, and the default value was changed to CPU. Please run with `--use-directml`.

**Reply by pw405** [reply]
Added --use-directml  to my command line args and it seems to be using the GPU now.  (7900 XTX).

**Reply by Dshinhal** [reply]
I updated my version today with git pull and it was updated to 1.7.0 and my AMD gpu stopped working. I thought I couldn't make images anymore. but I see this and use (--use-directml) and it works again

**Reply by nhk75** [reply]
Can confirm, adding "--use-directml" fixed the issues, but had to also add "--reinstall-torch" once to get everything else working too. 7800 XT card on windows 11.

**Reply by micbusin** [reply]
Brand new install.
If I start it with  _webui.bat --use-directml --reinstall-torch_ I got the message:
File "C:\AI\stable-diffusion-webui\modules\launch_utils.py", line 384, in prepare_environment
    raise RuntimeError(
RuntimeError: Torch is not able to use GPU; add --skip-torch-cuda-test to COMMANDLINE_ARGS variable to disable this check

If I start it with _webui.bat --help | findstr directml_ ther's nothing. No option containing _directml_ string.

If I start it with  _webui.bat --use-directml --skip-torch-cuda-test_ got the following:
C:\AI\stable-diffusion-webui>webui.bat --use-directml --skip-torch-cuda-test
venv "C:\AI\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.7.0
**Launching Web UI with arguments: --use-directml --skip-torch-cuda-test --use-directml**
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
usage: launch.py [-h] [--update-all-extensions] [--skip-python-version-check] [--skip-torch-cuda-test] [--reinstall-xformers] [--reinstall-torch] [--update-check] [--test-server] [--log-startup] [--skip-prepare-environment]
                 [--skip-install] [--dump-sysinfo] [--loglevel LOGLEVEL] [--do-not-download-clip] [--data-dir DATA_DIR] [--config CONFIG] [--ckpt CKPT] [--ckpt-dir CKPT_DIR] [--vae-dir VAE_DIR] [--gfpgan-dir GFPGAN_DIR]
                 [--gfpgan-model GFPGAN_MODEL] [--no-half] [--no-half-vae] [--no-progressbar-hiding] [--max-batch-count MAX_BATCH_COUNT] [--embeddings-dir EMBEDDINGS_DIR]
                 [--textual-inversion-templates-dir TEXTUAL_INVERSION_TEMPLATES_DIR] [--hypernetwork-dir HYPERNETWORK_DIR] [--localizations-dir LOCALIZATIONS_DIR] [--allow-code] [--medvram] [--medvram-sdxl] [--lowvram] [--lowram]
                 [--always-batch-cond-uncond] [--unload-gfpgan] [--precision {full,autocast}] [--upcast-sampling] [--share] [--ngrok NGROK] [--ngrok-region NGROK_REGION] [--ngrok-options NGROK_OPTIONS]
                 [--enable-insecure-extension-access] [--codeformer-models-path CODEFORMER_MODELS_PATH] [--gfpgan-models-path GFPGAN_MODELS_PATH] [--esrgan-models-path ESRGAN_MODELS_PATH] [--bsrgan-models-path BSRGAN_MODELS_PATH]
                 [--realesrgan-models-path REALESRGAN_MODELS_PATH] [--clip-models-path CLIP_MODELS_PATH] [--xformers] [--force-enable-xformers] [--xformers-flash-attention] [--deepdanbooru] [--opt-split-attention]
                 [--opt-sub-quad-attention] [--sub-quad-q-chunk-size SUB_QUAD_Q_CHUNK_SIZE] [--sub-quad-kv-chunk-size SUB_QUAD_KV_CHUNK_SIZE] [--sub-quad-chunk-threshold SUB_QUAD_CHUNK_THRESHOLD] [--opt-split-attention-invokeai]
                 [--opt-split-attention-v1] [--opt-sdp-attention] [--opt-sdp-no-mem-attention] [--disable-opt-split-attention] [--disable-nan-check] [--use-cpu USE_CPU [USE_CPU ...]] [--use-ipex]
                 [--disable-model-loading-ram-optimization] [--listen] [--port PORT] [--show-negative-prompt] [--ui-config-file UI_CONFIG_FILE] [--hide-ui-dir-config] [--freeze-settings] [--ui-settings-file UI_SETTINGS_FILE]
                 [--gradio-debug] [--gradio-auth GRADIO_AUTH] [--gradio-auth-path GRADIO_AUTH_PATH] [--gradio-img2img-tool GRADIO_IMG2IMG_TOOL] [--gradio-inpaint-tool GRADIO_INPAINT_TOOL] [--gradio-allowed-path GRADIO_ALLOWED_PATH]
                 [--opt-channelslast] [--styles-file STYLES_FILE] [--autolaunch] [--theme THEME] [--use-textbox-seed] [--disable-console-progressbars] [--enable-console-prompts] [--vae-path VAE_PATH] [--disable-safe-unpickle] [--api]
                 [--api-auth API_AUTH] [--api-log] [--nowebui] [--ui-debug-mode] [--device-id DEVICE_ID] [--administrator] [--cors-allow-origins CORS_ALLOW_ORIGINS] [--cors-allow-origins-regex CORS_ALLOW_ORIGINS_REGEX]
                 [--tls-keyfile TLS_KEYFILE] [--tls-certfile TLS_CERTFILE] [--disable-tls-verify] [--server-name SERVER_NAME] [--gradio-queue] [--no-gradio-queue] [--skip-version-check] [--no-hashing] [--no-download-sd-model]
                 [--subpath SUBPATH] [--add-stop-route] [--api-server-stop] [--timeout-keep-alive TIMEOUT_KEEP_ALIVE] [--disable-all-extensions] [--disable-extra-extensions] [--skip-load-model-at-start]
                 [--ldsr-models-path LDSR_MODELS_PATH] [--lora-dir LORA_DIR] [--lyco-dir-backcompat LYCO_DIR_BACKCOMPAT] [--scunet-models-path SCUNET_MODELS_PATH] [--swinir-models-path SWINIR_MODELS_PATH]
**launch.py: error: unrecognized arguments: --use-directml --use-directml**

_C:\AI\stable-diffusion-webui>python launch.py --help | findstr directml_ > nothing, ther's no --use-directml option.

**Reply by lshqqytiger** [reply]
You cloned `AUTOMATIC1111/stable-diffusion-webui`. Not `lshqqytiger/stable-diffusion-webui-directml`. A1111's webui does not have DirectML support.

**Comment by Kupatov-v** [comment]
didn't help me(

> launch.py: error: unrecognized arguments: --use-directml

**A complete reinstallation helped, with this config**

--use-directml --reinstall-torch

**Reply by Dshinhal** [reply]
I had a new problem when formatting my pc and I couldn't invoke --use-directml (if not torch.dml.has float64) but this solved it. thank you

**Comment by GriZen83** [comment]
I decided to share my experiences with the 1.7 upgrade. The upgrade was a complete failure and I recommend that anyone should wipe and install anew. 
The new install failed but you can fix it, by reinstalling pytorch. And then you need to use --skip-torch-cuda-test --use-directml to get it to run.

However you have to use --precision full --no-half --no-half-vae at launch otherwise the img2img will give you errors. ultimate sd upscale gets you huge seams and squares with incorrect colors erros.

Otherwise 1.7 is a improvement in everyway, especially it fixed instability with black screens ands bsods and system hard crashes.
Speed is the same vs 1.6 so I recommend to upgrade to 1.7. 

So good job @lshqqytiger, well done.

Now I am looking forward to rocm but that is in the future (in windows at least).

**Reply by pw405** [reply]
What GPU are you in @GriZen83 ? After adding the -- use DirectML, mine was working fine (7900XTX)  However I wonder if Ultimate SD Upscale seams are a result of not using --precision full?  I had used --precision full for a long time but recently took it out to see if it helped performance (didn't seem to matter much).

**Reply by GriZen83** [reply]
I have 7900xtx Hellhound. 
It seems is likely to be caused by insufficent precision. But it may also be --no-half-vae since the colors are wrong on some with different tiles. An upscale from 512X768 to 1024X1024 6 tiles. 

Strangely Sdupscale seems to work correctly, but Ultimate SD upscale does not without  --precision full --no-half --no-half-vae. 
My orginal parameters was ---upcast-sampling but using the new parameters works better. 
And most importanly 1.7 is much more stable. It hasnt crashed yet :D

**Reply by pw405** [reply]
Ok cool, I'll have to do some tests with Ultimate SD Upscale to see if the seams issue is better the additional command line arguments.  I've really been impressed by the 4X Foolhardy Remacari Upscaler... It's a 3rd party Upscaler, if I can remember later I'll dig up where I found it.

**Comment by GriZen83** [comment]
I have found that 4x_NMKD-Siax_200k is all arpund even better, especially with realworld photos but i use it with pretty all content.
